arroyo-console/src/routes/connections/ConfigureProfile.tsx

```diff
@@ -0,0 +1,232 @@
+        status: 'success',
+        duration: 9000,
+        isClosable: true,
+      });
+    }
+  };
+
+  return (
+    <>
+      <Stack spacing={4} padding={4} maxW={800}>
+        <Heading size={'xx-small'}>Cluster config</Heading>
+
+        <Text fontSize={'sm'}>
+          Select an existing {connector} connection profile or create a new one
+        </Text>
+
+        <Stack spacing={4}>
+          {connections
+            .filter(c => c.connector == connector)
+            .map(c => (
+              <HStack key={c.id}>
+                <LinkBox
+                  borderRadius={8}
+                  w={'100%'}
+                  p={4}
+                  border={'1px solid #777'}
+                  _hover={{ bg: 'gray.600', borderColor: 'black' }}
+                >
+                  <LinkOverlay href={'#'} onClick={() => onSubmit(c.id)} />
+                  <HStack spacing={4}>
+                    <Heading size={'14px'}>{c.name}</Heading>
+                    <Text fontStyle={'italic'}>{c.description}</Text>
+                  </HStack>
+                </LinkBox>
+                <Spacer />
+                <Box>
+                  <HStack>
+                    <IconButton
+                      variant={'outline'}
+                      aria-label={'Info'}
+                      icon={<InfoOutlineIcon />}
+                      onClick={() => {
+                        setEditingProfile(c);
+                        onOpen();
+                      }}
+                    />
+
+                    <IconButton
+                      variant={'outline'}
+                      aria-label={'Delete'}
+                      icon={<DeleteIcon />}
+                      onClick={() => deleteProfile(c)}
+                    />
+                  </HStack>
+                </Box>
+              </HStack>
+            ))}
+        </Stack>
+      </Stack>
+
+      <ClusterViewerModal
+        isOpen={isOpen}
+        onClose={onClose}
+        profile={editingProfile}
+        connector={connector}
+        schema={schema}
+      />
+    </>
+  );
+};
+
+export const ConfigureProfile = ({
+  connector,
+  onSubmit,
+  state,
+  setState,
+}: {
+  connector: Connector;
+  onSubmit: () => void;
+  state: CreateConnectionState;
+  setState: (s: CreateConnectionState) => void;
+}) => {
+  let { connectionProfiles, connectionProfilesLoading, mutateConnectionProfiles } =
+    useConnectionProfiles();
+  const [creatingCluster, setCreatingCluster] = useState<boolean>(false);
+
+  if (connectionProfilesLoading) {
+    return <></>;
+  }
+
+  if (creatingCluster) {
+    return (
+      <Stack spacing={8}>
+        <Box>
+          <Button leftIcon={<ArrowBackIcon />} onClick={() => setCreatingCluster(false)}>
+            Select existing
+          </Button>
+        </Box>
+        <CreateProfile
+          connector={connector}
+          addConnectionProfile={c => {
+            mutateConnectionProfiles();
```

Comment:
I think it's cleaner to use the `useConnectionProfiles` hook in the components where it's needed rather than passing these things around.